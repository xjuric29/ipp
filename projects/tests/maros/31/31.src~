.IPPcode18
JUMP $$MAIN
LABEL length
CREATEFRAME
DEFVAR TF@s
POPS TF@s
DEFVAR TF@len
STRLEN TF@len TF@s
PUSHS TF@len
RETURN
PUSHS int@0
RETURN
LABEL substr
CREATEFRAME
DEFVAR TF@s
POPS TF@s
DEFVAR TF@i
POPS TF@i
DEFVAR TF@n
POPS TF@n
DEFVAR TF@strval
MOVE TF@strval string@
DEFVAR TF@cmpres
LT TF@cmpres TF@i int@1
JUMPIFEQ substr$loopend TF@cmpres bool@true
SUB TF@i TF@i int@1
DEFVAR TF@len
STRLEN TF@len TF@s
SUB TF@len TF@len TF@i
GT TF@cmpres TF@n TF@len
JUMPIFNEQ strcmp$chk1 TF@cmpres bool@true
MOVE TF@n TF@len
LABEL strcmp$chk1
LT TF@cmpres TF@n int@0
JUMPIFNEQ strcmp$chk2 TF@cmpres bool@true
MOVE TF@n TF@len
LABEL strcmp$chk2
DEFVAR TF@char
LABEL substr$loopstart
GT TF@cmpres TF@n int@0
JUMPIFNEQ substr$loopend TF@cmpres bool@true
SUB TF@n TF@n int@1
GETCHAR TF@char TF@s TF@i
CONCAT TF@strval TF@strval TF@char
ADD TF@i TF@i int@1
JUMP substr$loopstart
LABEL substr$loopend
PUSHS TF@strval
RETURN
PUSHS string@
RETURN
LABEL asc
CREATEFRAME
DEFVAR TF@s
POPS TF@s
DEFVAR TF@i
POPS TF@i
DEFVAR TF@len
STRLEN TF@len TF@s
DEFVAR TF@cmp
SUB TF@i TF@i int@1
LT TF@cmp TF@i TF@len
JUMPIFNEQ asc$cmpfail TF@cmp bool@true
LT TF@cmp TF@i int@0
JUMPIFNEQ asc$cmpfail TF@cmp bool@false
DEFVAR TF@ascval
STRI2INT TF@ascval TF@s TF@i
PUSHS TF@ascval
RETURN
LABEL asc$cmpfail
PUSHS int@0
RETURN
PUSHS int@0
RETURN
LABEL chr
CREATEFRAME
DEFVAR TF@i
POPS TF@i
DEFVAR TF@s
INT2CHAR TF@s TF@i
PUSHS TF@s
RETURN
PUSHS string@
RETURN
LABEL $$MAIN
CREATEFRAME
DEFVAR TF@a
MOVE TF@a string@
MOVE TF@a string@ahoj
DEFVAR TF@$T0
LT TF@$T0 TF@a string@bhoj
JUMPIFNEQ $L2 TF@$T0 bool@true
WRITE string@OK
JUMP $L1
LABEL $L2
WRITE string@NOTOK
LABEL $L1
DEFVAR TF@$T3
EQ TF@$T3 TF@a string@ahoj
EQ TF@$T3 TF@$T3 bool@false
JUMPIFNEQ $L5 TF@$T3 bool@true
WRITE string@OK
JUMP $L4
LABEL $L5
WRITE string@NOTOK
LABEL $L4
DEFVAR TF@$T6
EQ TF@$T6 TF@a string@bhoj
EQ TF@$T6 TF@$T6 bool@false
JUMPIFNEQ $L8 TF@$T6 bool@true
WRITE string@OK
JUMP $L7
LABEL $L8
WRITE string@NOTOK
LABEL $L7
DEFVAR TF@$T9
LT TF@$T9 TF@a string@ahoj
EQ TF@$T9 TF@$T9 bool@false
JUMPIFNEQ $L11 TF@$T9 bool@true
WRITE string@OK
JUMP $L01
LABEL $L11
WRITE string@NOTOK
LABEL $L01
DEFVAR TF@$T21
LT TF@$T21 TF@a string@hoj
EQ TF@$T21 TF@$T21 bool@false
JUMPIFNEQ $L41 TF@$T21 bool@true
WRITE string@OK
JUMP $L31
LABEL $L41
WRITE string@NOTOK
LABEL $L31
DEFVAR TF@$T51
GT TF@$T51 TF@a string@ahoj
EQ TF@$T51 TF@$T51 bool@false
JUMPIFNEQ $L71 TF@$T51 bool@true
WRITE string@OK
JUMP $L61
LABEL $L71
WRITE string@NOTOK
LABEL $L61
DEFVAR TF@$T81
EQ TF@$T81 TF@a string@ahoj
JUMPIFNEQ $L02 TF@$T81 bool@true
WRITE string@OK
JUMP $L91
LABEL $L02
WRITE string@NOTOK
LABEL $L91
DEFVAR TF@$T12
EQ TF@$T12 TF@a string@bhoj
JUMPIFNEQ $L32 TF@$T12 bool@true
WRITE string@OK
JUMP $L22
LABEL $L32
WRITE string@NOTOK
LABEL $L22
