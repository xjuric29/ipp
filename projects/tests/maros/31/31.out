OK
NOTOK
OK
OK
NOTOK
OK
OK
NOTOK
